### PR TITLE
auto read notification

### PR DESCRIPTION
**Pull request by smallst**
fix eng-200
readfull can read the announcement
scroll to un-truncated announcement would read it using [IntersectionObserver](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API)

### CODE REVIEW

File: app/views/announcement/AnnouncementView.vue [hunk 1]
```diff
@@ -61,17 +66,50 @@ export default {
...
     }
   },
+  created () {
+    const options = {
+      root: this.$el,
+      rootMargin: '0px',
+      threshold: 1.0
+    }
+
+    this.observer = window.newIntersectionObserver(this.announcementShowUp, options)
```
Comment (mrfinch) [inline app/views/announcement/AnnouncementView.vue:76]: newIntersectionObserver?

Comment (smallst) [reply app/views/announcement/AnnouncementView.vue:76]: good catch

**Review by mrfinch — COMMENTED** [review]
I scrolled to the end of the page but it still showed 5 unread notifications. That isn't expected, correct?

**Comment by smallst** [comment]
Are 5 of them all long notifications? Long one only manually allow read by click read more( full content shown)  If so, expected. Otherwise maybe settimeout make things strange.. I can check again tomorrowSent from my phone-------- Original message --------From: Saurabh Patel ***@***.***>Date: Tue, Feb 13, 2024, 9:58 PMTo: codecombat/codecombat ***@***.***>Cc: Tong Yuqiang ***@***.***>, Author ***@***.***>Subject: Re: [codecombat/codecombat] auto read notification (PR #7255)
@mrfinch commented on this pull request.

I scrolled to the end of the page but it still showed 5 unread notifications. That isn't expected, correct?

**Review by mrfinch — APPROVED** [review]

**Pull request merged**